lapack/gonum/errors.go

```diff
@@ -6,134 +6,163 @@ package gonum
-	badVn2        = "lapack: vn2 has insufficient length"
-	badUplo       = "lapack: illegal triangle"
-	badWork       = "lapack: insufficient working memory"
-	badZ          = "lapack: insufficient z length"
-	zeroIncV      = "lapack: incv == 0"
-	badIncX       = "lapack: incX <= 0"
-	badIncY       = "lapack: incY <= 0"
-	i0LT0         = "lapack: i0 < 0"
-	n0LT0         = "lapack: n0 < 0"
-	kGTM          = "lapack: k > m"
-	kGTN          = "lapack: k > n"
-	kLT0          = "lapack: k < 0"
-	kLT1          = "lapack: k < 1"
-	kdLT0         = "lapack: kd < 0"
-	mLT0          = "lapack: m < 0"
-	mLTN          = "lapack: m < n"
-	mmLT0         = "lapack: mm < 0"
-	nGTM          = "lapack: n > m"
-	mGTN          = "lapack: m > n"
-	nanScale      = "lapack: NaN scale factor"
-	negANorm      = "lapack: anorm < 0"
-	negDimension  = "lapack: negative matrix dimension"
-	negZ          = "lapack: negative z value"
-	nLT0          = "lapack: n < 0"
-	nLT1          = "lapack: n < 1"
-	nbLT0         = "lapack: nb < 0"
-	nLTM          = "lapack: n < m"
-	nrhsLT0       = "lapack: nrhs < 0"
-	ncvtLT0       = "lapack: ncvt < 0"
-	nruLT0        = "lapack: nru < 0"
-	nccLT0        = "lapack: ncc < 0"
-	pLT0          = "lapack: p < 0"
-	offsetLT0     = "lapack: offset < 0"
-	offsetGTM     = "lapack: offset > m"
-	shortA        = "lapack: insufficient length of a"
-	shortAB       = "lapack: insufficient length of ab"
-	shortB        = "lapack: insufficient length of b"
-	shortC        = "lapack: insufficient length of c"
-	shortCNorm    = "lapack: insufficient length of cnorm"
-	shortD        = "lapack: insufficient length of d"
-	shortE        = "lapack: insufficient length of e"
-	shortF        = "lapack: insufficient length of f"
-	shortH        = "lapack: insufficient length of h"
-	shortIsgn     = "lapack: insufficient length of isgn"
-	shortQ        = "lapack: insufficient length of q"
-	shortS        = "lapack: insufficient length of s"
-	shortScale    = "lapack: insufficient length of scale"
-	shortT        = "lapack: insufficient length of t"
-	shortU        = "lapack: insufficient length of u"
-	shortV        = "lapack: insufficient length of v"
-	shortVL       = "lapack: insufficient length of vl"
-	shortVR       = "lapack: insufficient length of vr"
-	shortVT       = "lapack: insufficient length of vt"
-	shortW        = "lapack: insufficient length of w"
-	shortWH       = "lapack: insufficient length of wh"
-	shortWI       = "lapack: insufficient length of wi"
-	shortWR       = "lapack: insufficient length of wr"
-	shortWV       = "lapack: insufficient length of wv"
-	shortX        = "lapack: insufficient length of x"
-	shortY        = "lapack: insufficient length of y"
-	shortZ        = "lapack: insufficient length of z"
-	shortIWork    = "lapack: insufficient length of iwork"
-	shortWork     = "lapack: working array shorter than declared"
-	zeroDiv       = "lapack: zero divisor"
-
+	badApplyOrtho      = "lapack: bad ApplyOrtho"
```

Comment:
What is the ordering used here?

Reply:
The first block are bad enum values, then bad general parameter values, bad vector increments, bad slice lengths, insufficient slice lengths, and bad leading dimensions. The order of the blocks is arbitrary.

Within a block the strings are ordered alphabetically.

Reply:
We should probably have a comments saying what each block is so that someone coming later doesn't just place new things wherever.

Reply:
Actually I had comments for each block but then decided against them. I've put them back now.